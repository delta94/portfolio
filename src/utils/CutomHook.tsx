import { ChangeEvent, useState } from 'react';

const useForm = (initialValue: any) => {
  const [values, setValues] = useState(initialValue);
  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setValues({
      ...values,
      [e.target.name]: e.target.value,
    });
  };

  const reset = (name: string) =>
    setValues({
      ...values,
      [name]: '',
    });
  return [values, handleChange, reset];
};

export { useForm };
