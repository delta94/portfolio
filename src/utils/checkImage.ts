const checkImage = (path: string) => {
  new Promise((resolve, reject) => {
    const img = new Image();
    img.onload = () => resolve(path);
    img.onerror = () => reject();
    img.src = path;
    return {
      path: img.src,
      load: img.onload,
    };
  });
};

export { checkImage };
