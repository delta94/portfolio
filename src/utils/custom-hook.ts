import { ChangeEvent, useMemo, useState } from 'react';

const useForm = (initialValue: any) => {
  const [values, setValues] = useState(initialValue);
  const handleChange = (e: ChangeEvent<HTMLInputElement>) => {
    setValues({
      ...values,
      [e.target.name]: e.target.value,
    });
  };

  const reset = (name: string) =>
    setValues({
      ...values,
      [name]: '',
    });
  return [values, handleChange, reset];
};

const MemoizedComponent = (Component: JSX.Element, dependencies?: unknown) => {
  // eslint-disable-next-line
  return useMemo(() => Component, [Component, dependencies]);
};

export { useForm, MemoizedComponent };
