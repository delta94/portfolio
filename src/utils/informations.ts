export interface Information {
  startYear?: string;
  endYear?: string;
  nameOfOrganization?: string;
  details?: string;
}

export interface SkillInformationTypes {
  name?: string;
  progress?: number;
  backgroundColor?: string;
  technology?: 'BACKEND' | 'FRONTEND' | 'MISC';
}

export interface PhotoType {
  url: string;
}

export interface AvatarType {
  avatarOne: string;
  avatarTwo: string;
}
export interface Offerings {
  name: string;
  offerings: string[];
}

export interface MyOffersType {
  frontEnd: Offerings;
  backEnd: Offerings;
}

export interface Informations {
  avatar: AvatarType;
  educationInformations: Information[];
  experienceInformations: Information[];
  photos: PhotoType[];
  skillsInformation: SkillInformationTypes[];
  myOffers: MyOffersType;
}
