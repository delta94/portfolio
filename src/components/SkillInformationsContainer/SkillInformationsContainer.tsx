import clsx from 'clsx';
import React from 'react';
import { createUseStyles } from 'react-jss';
import { useMediaQuery } from 'react-responsive';
import { ProgressBarContainer } from '..';
import { CustomTheme } from '../../theme';
import { SkillInformationTypes } from '../../utils/informations';

export interface SkillInformationsContainerProps {
  skillsInformations?: SkillInformationTypes[];
}

const useStyles = createUseStyles((theme: CustomTheme) => ({
  mainContainer: {
    display: 'flex',
    justifyContent: 'space-evenly',
  },
  mainContainerMobile: {
    flexDirection: 'column',
  },
  textContainer: {
    borderBottomColor: theme.colors.thirdColor,
    borderBottomWidth: 2,
    borderBottomStyle: 'solid',
    paddingBottom: theme.spacing(2),
    paddingTop: theme.spacing(2),
  },
  infoContainer: {
    flex: 1,
    padding: theme.spacing(1),
  },
}));

type Technology = 'BACKEND' | 'FRONTEND' | 'MISC';

const getTechnology = (informations: SkillInformationTypes[], technology: Technology) =>
  informations.filter((informations: SkillInformationTypes) => informations.technology === technology);

const technologies: Technology[] = ['FRONTEND', 'BACKEND', 'MISC'];

const SkillInformationsContainer: React.FC<SkillInformationsContainerProps> = ({ skillsInformations }) => {
  const classes: Record<
    'mainContainer' | 'textContainer' | 'infoContainer' | 'mainContainerMobile',
    string
  > = useStyles();

  const isMobile: boolean = useMediaQuery({ maxWidth: 780 });

  return (
    <>
      {skillsInformations !== undefined && (
        <div className={clsx([classes.mainContainer, isMobile && classes.mainContainerMobile])}>
          {technologies.map((technology: Technology) => {
            return (
              <div key={technology} className={classes.infoContainer}>
                <h4 className={classes.textContainer}>{technology}</h4>
                {getTechnology(skillsInformations, technology).map((skillInformation: SkillInformationTypes) => {
                  return (
                    <ProgressBarContainer
                      name={skillInformation.name}
                      progress={skillInformation.progress}
                      backgroundColor={skillInformation.backgroundColor}
                      key={skillInformation.name}
                    />
                  );
                })}
              </div>
            );
          })}
        </div>
      )}
    </>
  );
};

export { SkillInformationsContainer };
