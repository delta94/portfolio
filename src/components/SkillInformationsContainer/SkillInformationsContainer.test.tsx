import { render } from '@testing-library/react';
import React from 'react';
import { TestContextProvider } from '../../TestContextProvider';
import { SkillInformationTypes } from '../../utils/informations';
import { SkillInformationsContainer } from './SkillInformationsContainer';

const mockData = {
  skillsInformations: [{ name: 'mock name', backgroundColor: 'blue', progress: 50, technology: 'BACKEND' }],
};
test('SkillInformationsContainer renders correctly', () => {
  const { asFragment } = render(
    <TestContextProvider>
      <SkillInformationsContainer skillsInformations={mockData.skillsInformations as SkillInformationTypes[]} />
    </TestContextProvider>,
  );
  expect(asFragment()).toMatchSnapshot();
});
