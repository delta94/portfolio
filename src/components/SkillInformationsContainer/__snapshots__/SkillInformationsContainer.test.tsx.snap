// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SkillInformationsContainer renders correctly 1`] = `
<DocumentFragment>
  <div
    class="mainContainer-0-2-1"
  >
    <div
      class="infoContainer-0-2-4"
    >
      <h4
        class="textContainer-0-2-3"
      >
        FRONTEND
      </h4>
    </div>
    <div
      class="infoContainer-0-2-4"
    >
      <h4
        class="textContainer-0-2-3"
      >
        BACKEND
      </h4>
      <div
        class="progressBarParent-0-2-9"
      >
        <div
          class="description-0-2-6"
        >
          <h3
            class="text-0-2-10"
          >
            mock name
          </h3>
          <h3
            class="text-0-2-10"
          >
            50%
          </h3>
        </div>
        <div
          class="progressBarContainer-0-2-7"
        >
          <div
            class="progressBar-0-2-8 progressBar-d0-0-2-11"
          />
        </div>
      </div>
    </div>
    <div
      class="infoContainer-0-2-4"
    >
      <h4
        class="textContainer-0-2-3"
      >
        MISC
      </h4>
    </div>
  </div>
</DocumentFragment>
`;
